modules/svg/data.js

```diff
@@ -427,6 +444,45 @@ export function svgData(projection, context, dispatch) {
         return this;
     };
 
+    function geometryCollectionToFeatures(gj) {
+        var currentFeatures = getFeatures(gj);
+
+        var newFeaturesList = _reduce(currentFeatures, function(featuresAggregate, featureItem) {
+            var features = [];
+
+            if (featureItem.geometry.type === 'GeometryCollection') {
+                var geometries = featureItem.geometry.geometries;
+
+                geometries.forEach( function(geom, index) {
+                    var newFeature = _cloneDeep(featureItem);
+                    newFeature.roles = newFeature.roles.split(';')[index];
```

Comment:
So `roles` isn't really a valid geojson property and neither is `relation-type`.  Not sure we need to handle it explicitly.  Rather than a deep clone maybe just copy the `properties` property to all output features.